### PR TITLE
Add Hdf5 file I/O library from Anboto to registry

### PR DESCRIPTION
**Pull request by klugier**

### CODE REVIEW

**Comment by ismail-yilmaz** [comment]
Package registry is OK for UppHub.

On linux:

- Compiles fine on GCC
- Fails to compile with CLANG

Errors:



`linux kernel version: 6.6.2`
clang version 16.0.6, Target: x86_64-pc-linux-gnu, Thread model: posix`


Generates a lot of similar error:

`
/home/testusr/upp/.config/u++/theide/UppHub/Hdf5-main/plugin/Hdf5/src/H5Defl.c (266): error: call to undeclared function 'lseek'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
`

File: nests.json [hunk 1]
```diff
@@ -201,6 +201,18 @@
       "category": "scientific",
       "readme": "https://raw.githubusercontent.com/anboto/Assimp/main/README.md"
     },
+    {
+      "name": "Hdf5",
+      "packages": [
+        "Hdf5",
+        "plugin/Hdf5"
+      ],
+      "description": "HDF5 Hierarchical Data Format (HDF) library wrapper ",
```
Comment (ismail-yilmaz) [inline nests.json:210]: Purely cosmetics: There is a space at the end of the string. It can be removed.

File: nests.json [hunk 2]
```diff
@@ -201,6 +201,18 @@
...
+      "repository": "https://github.com/anboto/Hdf5",
```
Comment (ismail-yilmaz) [inline nests.json:211]: Shouldn't it be "https://github.com/anboto/Hdf5.git" ?

**Review by ismail-yilmaz — APPROVED** [review]
OK, seems all fine now.

@izabala123 might want to look into CLANG errors on Linux. On GCC everything seems fine.

**Comment by izabala123** [comment]
Hi Ismail

Thank you for reporting this problem.
I have tried to reproduce it unsuccessfully using GCC and CLANG, debug and release. 
Maybe did it appear in a previous version? I have done many improvements the previous days.
Please I would like to ask you to check it again.
My Linux version is Ubuntu 22.04.3 LTS Jammy. Kernel version is 5.15.90.4. CLANG version is 14.0.0

Best regards
Iñaki

**Comment by izabala123** [comment]
Hi Ismail

I have upgraded CLANG until version 17 and I have reproduced your errors. 
They are now fixed, and the changes also work with previous version 14.
My Linux version now is Ubuntu 23.04. Lunar. Kernel version is 5.15.90.4. CLANG version is 17.0.6.
Please update UppHub and check it again. 
Thank you!

Best regards
Iñaki

**Comment by ismail-yilmaz** [comment]
Hello Iñaki
Everything seems fine with the latest package. 

Best regards,
İsmail

**Pull request merged**